### PR TITLE
[RISCV] Add Qualcomm uC Xqcia (Arithmetic) extension

### PR DESCRIPTION
**Pull request by svs-quic**
This extension adds 11 instructions that perform integer arithmetic.

The current spec can be found at:
https://github.com/quic/riscv-unified-db/releases/latest

This patch adds assembler only support.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVInstrInfoXqci.td [hunk 1]
```diff
@@ -72,3 +84,27 @@ let Predicates = [HasVendorXqcisls, IsRV32], DecoderNamespace = "Xqcisls" in {
...
+  def QC_WRAPI : RVInstI<0b000, OPC_CUSTOM_0, (outs GPRNoX0:$rd),
+                         (ins GPRNoX0:$rs1, uimm11:$imm11), "qc.wrapi",
+                         "$rd, $rs1, $imm11"> {
+    bits<11> imm11;
+
+    let imm12 = {0b0, imm11};
+  }
+
+  def QC_NORM : QCIRVInstR<0b0111, "qc.norm">;
+  def QC_NORMU : QCIRVInstR<0b1000, "qc.normu">;
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVInstrInfoXqci.td:107]: I have a question about the semantics of qc.normu in the spec

```
XReg clz = (xlen() - 1) - $signed(highest_set_bit(X[rs1]));
XReg exp = (X[rs1] << clz);
XReg mnt = (-clz);
X[rd] = {mnt[23:0], exp[7:0]};
```

Are the assignments of `mnt` and `exp` swapped? Unless I'm misunderstanding, I would expect `exp` to be based only on `clz` and `mnt` to be baed on `(X[rs1] << clz)`.

Also the assignment of `X[rd] = {mnt[23:0], exp[7:0]};` seems like it should be taking the upper 24 bits of `mnt` not the lower 24 bits? But maybe I've misunderstood?

**Review by lenary — APPROVED** [review]
LGTM.

I note Craig has questions about the spec, we will ensure those are relayed to the spec authors, but they don't relate to encoding/assembling problems, so I'm not sure if they should block landing this.

**Comment by topperc** [comment]
> LGTM.
> 
> 
> 
> I note Craig has questions about the spec, we will ensure those are relayed to the spec authors, but they don't relate to encoding/assembling problems, so I'm not sure if they should block landing this.

I don't consider my questions blocking. I just didn't know where else to ask.

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**